content/Rails - Multiple Databases.md

```diff
@@ -0,0 +1,124 @@
+# ** Rails 6 - Multiple Databases **
+
+![main image](images/databases.jpg)
+
+Rails 6.0 came up with this amazing enhancement, but for most people, I'm aware they think that is a new feature.
+In my opinion, both are correct, because the actual state of multiple databases before rails 6.0 was not even useful to consider it a completed feature.
+
+Let's dive into Rails < 6 state, Rails 6 introduced changes and the current roadmap.
+
+
+## Rails 5 state
+ActiveRecord supports multiple databases, but Rails < 6 doesn't provide a way to manage them.
+As said you needed to handle everything (yes, everything) in a really manual way, having to create your own tasks for `seeds`, `db:prepare`, `db:migrate`, etc.
+Then you needed to create a sort of generator to handle your secondary database migrations following conventions. And finally, as if you hadn't done enough custom work, create an initializer to config your connection to the new database.
+
+So yes, ActiveRecord supported multiple databases, but it was a complete and real mess you didn't want to get into unless really necessary.
+
+## Rails 6.0 state
+
+#### New stuff
+
+- Multiple primary databases and a replica for each
+
+This is amazing, now you can easiliy configure it in your `database.yml` like this
+``` ruby
+development:
+  primary:
+    database: my_primary_database
+    user: root
+    adapter: sqlite3
+  primary_replica:
+    database: my_primary_database
+    user: root_readonly
+    adapter: sqlite3
+    replica: true
+  animals:
+    database: my_animals_database
+    user: animals_root
+    adapter: sqlite3
+    migrations_paths: db/animals_migrate
+  animals_replica:
+    database: my_animals_database
+    user: animals_readonly
+    adapter: sqlite3
+    replica: true
+```
+
```

Comment:
Another thing that might be worth adding is how easy is to fine-grain configure our models to use these multiple DBs using `connect_to`, e.g.:

```ruby
connects_to database: { writing: :some_primary_db, reading: :some_primary_db_replica }
```